File: py-tritium-remote/python/start_script.py
import os
import asyncio
import tritium_remote

SCRIPT_PATH = "start_stop.py"


async def main():
    auth_token = os.environ["TRITIUM_AUTH_TOKEN"]
    host = os.environ.get("TRITIUM_HOST", "localhost")

    print("connecting...")
    tritium = await tritium_remote.connect(f"ws://{host}:1234", auth_token, , "Python tritium-remote example - start script")

    print(f"starting script {SCRIPT_PATH}...")
    await tritium.start_script(SCRIPT_PATH)
    print("done.")


asyncio.run(main())

File: tritium-remote/examples/start_script.rs
use simple_logger::SimpleLogger;
use std::env;

const SCRIPT_PATH: &str = "start_stop.py";

#[tokio::main]
async fn main() {
    // logging controlled by the RUST_LOG environment variable
    SimpleLogger::new().env().init().unwrap();

    let host = env::var("TRITIUM_HOST").unwrap_or("localhost".to_string());

    let auth_token =
        env::var("TRITIUM_AUTH_TOKEN").expect("TRITIUM_AUTH_TOKEN environment variable not set");

    let url = format!("ws://{host}:1234");

    let mut tritium = tritium_remote::connect(
        &url,
        &auth_token,
        Some("tritium-remote example - start script".to_string()),
    )
    .await
    .expect("failed to connect");

    log::info!("Starting script {}", SCRIPT_PATH);
    let script = tritium
        .start_script(SCRIPT_PATH)
        .await
        .expect("trigger script mutation failed");

    log::info!("OK, start_script returned: {:?}", script);
}
